broker-core/src/main/java/org/apache/qpid/server/security/group/FileGroupDatabase.java

```diff
@@ -174,10 +179,27 @@ public synchronized void removeUserFromGroup(String user, String group)
     @Override
     public synchronized void createGroup(String group)
     {
-        Set<String> users = new ConcurrentSkipListSet<String>();
-        _groupToUserMap.put(group, users);
+        if (!isAvailable(group, _groupToUserMap))
+        {
+            Set<String> users = new ConcurrentSkipListSet<String>();
+            _groupToUserMap.put(group, users);
+            update();
+        }
+        else
+        {
+            throw new IllegalConfigurationException(String.format("Group with name '%s' already exists", group));
+        }
+    }
 
-        update();
+    private boolean isAvailable(String searchString,
+                                final Map<String, Set<String>> collection)
+    {
+        String groupAvailable = keySearch(collection.keySet(), searchString);
+        if (collection.get(groupAvailable) == null && groupAvailable == searchString)
```

Comment:
Please replace this method implementation with below

	private boolean exists(final String searchString, final Set<String> set)
	{
		if (!_groupProvider.isCaseSensitive())
		{
		    for (String key : set)
		    {
			if (key.equalsIgnoreCase(searchString))
			{
			    return true;
			}
		    }
		    return false
		}
		return set.contains(searchString);
	}

Reply:
Done!